### PR TITLE
Fixed always default error handling

### LINKED ISSUE #17: ErrorHandler decorator
The documentation mentions using the `@ErrorHandler()` decorator, but that decorator doesn't exist.

### PR DESCRIPTION
**Pull request by 19majkel94**
Fixed #17 and #23.

Repaired executing default error handling even when it's turned off, which made impossible defining own error handler.

Exported `HttpError` to simplify using default error handler by throwing error in controller:
```typescript
@Post("/login")
async loginUser(@Session() session: Express.Session, @Body({ required: true }) bodyUser: BodyUser) {
    if (typeof bodyUser.email !== "string" || typeof bodyUser.password !== "string") {
        throw new HttpError(400, "The email and password fields must be string!");
    }

    const dbUser = await this.userRepository.getByEmail(bodyUser.email);
    if (dbUser && await dbUser.verifyPasswordAsync(bodyUser.password)) {
        session.user = dbUser;
        return "Logged succesfully!";
    } else {
        throw new HttpError(400, "Incorrect email or password!");
    }
}
```

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -121,6 +121,7 @@ export * from "./decorator/controllers";
 export * from "./decorator/decorators";
 export * from "./decorator/methods";
 export * from "./decorator/params";
+export * from "./error/http/HttpError";
```
Comment (pleerock) [inline src/index.ts]: I think we can add export for all other errors too, to make use of them convenient too

**Comment by pleerock** [comment]
Can you please add tests for the issues this PR resolves?

File: src/index.ts (hunk 1 above)
Comment (19majkel94) [inline src/index.ts]: You are right! I've done it in 6d16e78.
Now you can import them also this way:
`import { HttpError } from "routing-controllers/error";`

**Comment by 19majkel94** [comment]
> Can you please add tests for the issues this PR resolves?

It's hard to easy build a test in your `express-error-handling.spec.ts` - I need to disable default error handler in `createExpressServer` options and register new `MiddlewareGlobalAfter`, which might break your other tests. I would have to create a new test file for this case, is that ok for you?

**Comment by pleerock** [comment]
try the best possible solution

**Comment by pleerock** [comment]
released in 0.6.6

**Pull request merged**